ksqldb-common/src/main/java/io/confluent/ksql/errors/LogMetricAndContinueExceptionHandler.java

```diff
@@ -39,15 +41,40 @@ public DeserializationHandlerResponse handle(
   ) {
     log.debug(
         "Exception caught during Deserialization, "
-        + "taskId: {}, topic: {}, partition: {}, offset: {}",
+            + "taskId: {}, topic: {}, partition: {}, offset: {}",
         context.taskId(), record.topic(), record.partition(), record.offset(),
         exception
     );
 
     streamsErrorCollector.recordError(record.topic());
+
+    final OptionalLong restErrorCode = recursiveFindErrorCode(exception, 10);
+
+    if (restErrorCode.isPresent() && restErrorCode.getAsLong() == 40301) {
```

Comment:
I find a method `isForbiddenHttpStatusError(exception)` more useful when looking at this exception. That way you hide the implementation on how to figure out whether the exception is a RestClientException and if it has the 403 error code.

Reply:
Good idea, done